R/data-storage-log-file.R

```diff
@@ -0,0 +1,185 @@
+
+    read_session_data = function(date_from, date_to) {
+      db_data <- private$read_data(private$session_file_path, date_from, date_to)
+
+      db_data %>%
+        dplyr::select("session", "detail") %>%
+        dplyr::group_by(.data$session) %>%
+        dplyr::summarise(title = paste(.data$detail, collapse = " | "))
+    },
+
+    #' @description read all session data
+
+    close = function() {
+      private$close_connection()
+    }
+  ),
+  active = list(
+
+    #' @field action_bucket string that identifies the file path to store user
+    #' related and action data
+
+    action_bucket = function() {
+      private$log_file_path
+    },
+
+    #' @field session_bucket string that identifies the bucket to store session
+    #' details data
+
+    session_bucket = function() {
+      private$session_file_path
+    }
+  ),
+  #
+  # Private
+  private = list(
+    # Private fields
+    log_file_path = NULL,
+
+    session_file_path = NULL,
+
+    # Private methods
+
+    # @name connect
+    # Makes connection to database based on passed config data
+    # @param log_file_path string with path to file for user actions
+    # @param session_file_path string with path to file for session details
+
+    connect = function(log_file_path, session_file_path) {
+      private$log_file_path <- log_file_path
+      private$session_file_path <- session_file_path
+    },
+
+    # @description reverts logger settings to default
+
+    close_connection = function() {
+    },
+
+    insert_checks = function(values, add_username) {
+      if ("time" %in% names(values)) {
+        rlang::abort(paste0(
+          "You must not pass 'time' value into database.",
+          " It is set automatically."
+        ))
+      }
+
+      if ("session" %in% names(values)) {
+        rlang::abort(paste0(
+          "You must not pass 'session' value into database.",
+          " It is set automatically."
+        ))
+      }
+
+      if ("username" %in% names(values)) {
+        rlang::abort(paste0(
+          "You must not pass 'username' value into database.",
+          " It is set automatically."
+        ))
+      }
+
+      values$time <- Sys.time()
+      values$session <- private$.session_id
+
+      if (add_username) {
+        values$username <- private$.username
+      }
+
+      values
+    },
+
+    write = function(values, bucket) {
+      cat(jsonlite::toJSON(values), file = bucket, sep = "\n", append = TRUE)
+    },
+
+    # @name read_data
+    # Reads the JSON log file
+    # @param bucket string with path to file
+    read_data = function(bucket, date_from, date_to) {
+      checkmate::assert_string(bucket)
+      checkmate::assert_date(date_from)
+      checkmate::assert_date(date_to)
+
+      json_log_msg <- readLines(bucket)
+      json_log <- dplyr::bind_rows(lapply(json_log_msg, jsonlite::fromJSON))
+      json_log %>%
+        dplyr::filter(
+          time >=  date_from,
+          time <= date_to
+        )
+    }
```

Comment:
One of the main downsides of using a log file. We really cannot query the file and filter it before reading. Here we read all the file then filter it based on the specified date range